sycl/include/sycl/handler.hpp

```diff
@@ -28,6 +28,7 @@
 #include <sycl/exception.hpp>
 #include <sycl/ext/intel/experimental/fp_control_kernel_properties.hpp>
 #include <sycl/ext/intel/experimental/kernel_execution_properties.hpp>
+#include <sycl/ext/intel/experimental/virtual_functions.hpp>
```

Comment:
Note for reviewers: I've tried not to include this here, but a bigger refactoring is needed if we won't want to see includes like this in `handler.hpp`.

The problem is that `property_list::has_property` depends on many type traits which are scattered around several files and property-specific header files like the one I'm adding specialize them. It is not immediately clear to me how to re-organize the code without having to split an implementation of properties for an extension into several files.

As of right now `virtual_functions.hpp` is relatively small, but implementing a `FIXME` about proper handling of `calls_indirectly` property will bring a couple hundreds lines of template-heavy code and then the impact will be bigger. I suppose the refactoring should happen somewhere in between.